supervisor/iptablesctrl/uidpam/update_portsets.go

```diff
@@ -0,0 +1,260 @@
+package uipam
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/user"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aporeto-inc/trireme-lib/cache"
+	"github.com/bvandewalle/go-ipset/ipset"
+	"go.uber.org/zap"
+)
+
+const (
+	procNetTCPFile                    = "/proc/net/tcp"
+	portSetUpdateIntervalMilliseconds = 1000
+	portEntryTimeout                  = 60
+	uidFieldOffset                    = 7
+	procHeaderLineNum                 = 1
+	portOffset                        = 1
+	ipPortOffset                      = 1
+	sockStateOffset                   = 3
+	sockListeningState                = "0A"
+	hexFormat                         = 16
+	integerSize                       = 64
+	decFormat                         = 10
+)
+
+// PortSet : This provides an interface to update the
+// look up table required to program the ipset portsets.
+type PortSet interface {
+	AddToUIDToPortSetCache(uid string, value string) (err error)
```

Comment:
I see a lot more functions than what we talked about. I was expecting 

type PortSet interface {
    // AddUIDPortSet is called when a user is created
    AddUID(...)
    // DeleteUIDPortSet is called when a user is deleted
    DeleteUID(...)
    // AddUIDPort is called when a new port used by a user is detected
    AddPortToUID(...)
}